fury/ui.py

```diff
@@ -2341,22 +2386,71 @@ def set_position(self, position, disk_number):
         disk_number : int
             The index of disk being moved.
         """
-        x_position = position[0]
 
-        if disk_number == 0 and x_position >= self.handles[1].center[0]:
-            x_position = self.ratio_to_coord(
-                self.value_to_ratio(self._values[1] - 1))
+        if self.orientation == "horizontal":
+            x_position = position[0]
+
+            if disk_number == 0 and x_position >= self.handles[1].center[0]:
+                x_position = self.ratio_to_coord(
+                    self.value_to_ratio(self._values[1] - 1))
 
-        if disk_number == 1 and x_position <= self.handles[0].center[0]:
-            x_position = self.ratio_to_coord(
-                self.value_to_ratio(self._values[0] + 1))
+            if disk_number == 1 and x_position <= self.handles[0].center[0]:
```

Comment:
I do not understand, Can you explain to me when this case happens?

Reply:
This case happens when the right Disk tries to go beyond the center of the left disk. So each time the right disk tries to go beyond, I shift it to the right by 1 pixel so that it never goes beyond.

Reply:
ok, make sense